### ISSUE TITLE
Catalog.Iterator test failure

### ISSUE THREAD

**Issue by michael-loki** [loki-project/loki-network#468]
```
Repeating all tests (iteration 779) . . .

Note: Google Test filter = Catalog.Iterator
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from Catalog
[ RUN      ] Catalog.Iterator
../test/util/test_llarp_util_object.cpp:92: Failure
Value of: catalog->find(h)
  Actual: true
```

This looks like we end up getting duplicate handles on occasion. In this failure, 2 threads got id `-1266679808`.

**Comment by michael-loki** [comment]
how tf this happens, I have no idea.

**Comment by majestrate** [comment]
probably by the power of satan.

**Comment by michael-loki** [comment]
might make everything `uint32_t`:

```
../llarp/util/object.hpp:141:24: runtime error: signed integer overflow: 2139095048 + 16777216 cannot be represented in type 'int'
```

**Comment by michael-loki** [comment]
this was fixed in #472
